src/broadcaster.rs

```diff
@@ -0,0 +1,192 @@
+use std::sync::{Arc, Mutex, Weak};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use futures_core::Async;
+use futures_core::task::{Context, Waker, Wake};
+
+use super::signal::*;
+
+/// Wraps any Signal to make it Clone.
+pub struct Broadcaster<A: Signal>
```

Comment:
As explained on the other issue, the API should be like this:

```rust
let broadcaster = Broadcaster::new(some_signal);

broadcaster.signal()
broadcaster.signal_cloned()
```

This is consistent with `Mutable`, and it's also faster: if the `Signal` value is `Copy` (which is common with types like `u32`, `f64`, etc.) then it's better to use `Copy` rather than `Clone`

Reply:
Cool I'll make it like that 😄